core/trino-main/src/main/java/io/trino/sql/planner/iterative/rule/AdaptiveReorderPartitionedJoin.java

```diff
@@ -247,4 +235,98 @@ private static StreamProperties deriveStreamPropertiesRecursively(PlanNode node,
                 .collect(toImmutableList());
         return deriveStreamPropertiesWithoutActualProperties(resolvedNode, inputProperties, metadata, session);
     }
+
+    private static class BuildToProbeLocalExchangeRewriter
+            extends SimplePlanRewriter<Void>
+    {
+        private final PlanNodeId localExchangeNodeId;
+        private final Context context;
+
+        private BuildToProbeLocalExchangeRewriter(PlanNodeId localExchangeNodeId, Context context)
+        {
+            this.localExchangeNodeId = requireNonNull(localExchangeNodeId, "localExchangeNodeId is null");
+            this.context = requireNonNull(context, "context is null");
+        }
+
+        @Override
+        protected PlanNode visitPlan(PlanNode node, RewriteContext<Void> ctx)
+        {
+            return rewriteSources(this, node, context);
+        }
+
+        @Override
+        public PlanNode visitAggregation(AggregationNode node, RewriteContext<Void> ctx)
+        {
+            if (node.getStep() == PARTIAL) {
+                return rewriteSources(this, node, context);
+            }
```

Comment:
This is the same as `visitPlan(node, ctx);`.  Looks like you can just drop `if`. Or whole `visitAggregation` actually